### PR TITLE
SyntheaEDA Jupyter Notebooks - Added for DIG-757 and subtasks

### PR DESCRIPTION
**Pull request by AliRZ-02**
### DESCRIPTION
This pull request relates to the [DIG-757 Jira Ticket](https://candig.atlassian.net/browse/DIG-757) and is our first Jira Ticket during our time at UHN. The purpose of this project was to get antiquated with existing ML Libraries, as well as to get some experience with setting up development environments on our local machine. It's also good practice for when we work to integrate the federated-learning repo into the CanDIG-V2 stack. (Note, there was additional EDA done to explore and test inputs and outputs, which was omitted for clarity purposes).

The following parts of DIG-757 are completed by this PR:
- DIG-758: Install Katsu, Populate Katsu with Synthea Data, Familiarize oneself with API endpoints, etc.
- DIG-759: Uses Jupyter Notebooks to perform EDA of the collected Synthea Data
- DIG-759: Selects Set of inputs/outputs
- DIG-759: Selects default classifier to use (Decision Tree)
- DIG-760: Uses Jupyter Notebooks to document and train selected classifiers
- DIG-757: Creates README file in Ali_Laiba_syntheaEDA directory with important information regarding the EDA (eg. dependencies, etc.)

### CHANGELOG
- Added EDA Jupyter Notebooks: [SyntheaEDA](https://github.com/CanDIG/federated-learning/blob/AliRZ-02/DIG-757-SyntheaEDA/examples/synthea-breast-cancer/Ali_Laiba_syntheaEDA/SyntheaEDA.ipynb) and [SyntheaClassifiers](https://github.com/CanDIG/federated-learning/blob/AliRZ-02/DIG-757-SyntheaEDA/examples/synthea-breast-cancer/Ali_Laiba_syntheaEDA/SyntheaClassifiers.ipynb), partially documenting the EDA process for [DIG-759](https://candig.atlassian.net/browse/DIG-759) and [DIG-760](https://candig.atlassian.net/browse/DIG-760). These notebooks detail the process of selecting and processing data from the Synthea MCODE Synthetic Breast Cancer Data and also detail how the training of several different classifiers occurs using the `scikit-learn` module within Python. The classifiers have been tested and trained using the Synthea Data, and further documentation exists at the respective [confluence](https://candig.atlassian.net/wiki/spaces/CA/pages/624427043/Synthea+Breast+Cancer+Dataset+-+Inputs+and+Outputs) pages for [both](https://candig.atlassian.net/wiki/spaces/CA/pages/624623655/Synthea+Breast+Cancer+Dataset+-+Classifier+Training) of the EDA's.

### DEPENDENCIES
- Requires Local Katsu Instance with Synthea Data Ingested at port 8000
- Requires GraphQL-interface endpoint set up and directed to port 7999
- Requires updates to GraphQL repo CanDIG/GraphQL-interface#5

#### Quickstart 
- View the [README File](https://github.com/CanDIG/federated-learning/blob/AliRZ-02/DIG-757-SyntheaEDA/examples/synthea-breast-cancer/Ali_Laiba_syntheaEDA/README.md) for a Quickstart Guide

### CODE REVIEW

**Comment by amanjeev** [comment]
@AliRZ-02  @Zamm178  This PR has the best description I have seen in most my professional settings. I needed to say that before any review.

**Comment by amanjeev** [comment]
This work is all done with Jupyter notebooks? I wish github showed rendered notebooks.

**Comment by yuriy-pavlov** [comment]
@amanjeev 
> Just thinking out loud. -- This work is all done with Jupyter notebooks? I wish github showed rendered notebooks.

It does - if you navigate to the file using the repository's file tree (instead of loading the diff) it will render. Ex.: https://github.com/CanDIG/federated-learning/blob/AliRZ-02/DIG-757-SyntheaEDA/examples/synthea-breast-cancer/Ali_Laiba_syntheaEDA/SyntheaEDA.ipynb

File: examples/synthea-breast-cancer/Ali_Laiba_syntheaEDA/SyntheaEDA.ipynb
Comment (yuriy-pavlov) [inline examples/synthea-breast-cancer/Ali_Laiba_syntheaEDA/SyntheaEDA.ipynb:971]: I like that you saved the dataframe built in the EDA and reused it in the training notebook, very helpful.
In the future you might want to try out `pickling` instead of a csv to accomplish this, I don't know a ton about it but it's the more proper-data-science-y way of sharing data between notebooks. Here's an article on it: https://betterprogramming.pub/dont-fear-the-pickle-using-pickle-dump-and-pickle-load-5212f23dbbce

File: examples/synthea-breast-cancer/Ali_Laiba_syntheaEDA/SyntheaClassifiers.ipynb
Comment (yuriy-pavlov) [inline examples/synthea-breast-cancer/Ali_Laiba_syntheaEDA/SyntheaClassifiers.ipynb:382]: Not sure what is being accomplished in this block, could you add some narrative text to describe what this code does?

File: examples/synthea-breast-cancer/Ali_Laiba_syntheaEDA/SyntheaClassifiers.ipynb
Comment (yuriy-pavlov) [inline examples/synthea-breast-cancer/Ali_Laiba_syntheaEDA/SyntheaClassifiers.ipynb:406]: Generous and appreciated

File: examples/synthea-breast-cancer/Ali_Laiba_syntheaEDA/SyntheaClassifiers.ipynb
Comment (yuriy-pavlov) [inline examples/synthea-breast-cancer/Ali_Laiba_syntheaEDA/SyntheaClassifiers.ipynb:457]: Thank you so much for organizing this section with a table of contents. Very neat.

File: examples/synthea-breast-cancer/Ali_Laiba_syntheaEDA/SyntheaClassifiers.ipynb
Comment (yuriy-pavlov) [inline examples/synthea-breast-cancer/Ali_Laiba_syntheaEDA/SyntheaClassifiers.ipynb:763]: This section is really a great use of the Jupyter Notebook format, getting to walk through the code used to generate the model and visualize its internals is super cool.

File: examples/synthea-breast-cancer/Ali_Laiba_syntheaEDA/SyntheaClassifiers.ipynb
Comment (yuriy-pavlov) [inline examples/synthea-breast-cancer/Ali_Laiba_syntheaEDA/SyntheaClassifiers.ipynb:835]: Really enjoyed your discussion of the Decision Tree classifier both here and in the Confluence discussion. Great analysis, choice, and justifications.

File: examples/synthea-breast-cancer/Ali_Laiba_syntheaEDA/SyntheaClassifiers.ipynb
Comment (yuriy-pavlov) [inline examples/synthea-breast-cancer/Ali_Laiba_syntheaEDA/SyntheaClassifiers.ipynb:1260]: Super cool, thanks for putting in the effort to one-hot encode everything for this experiment

File: examples/synthea-breast-cancer/Ali_Laiba_syntheaEDA/SyntheaClassifiers.ipynb
Comment (yuriy-pavlov) [inline examples/synthea-breast-cancer/Ali_Laiba_syntheaEDA/SyntheaClassifiers.ipynb:1594]: Good points

**Review by yuriy-pavlov — CHANGES_REQUESTED** [review]
Beautiful work on the PR description and the supplementary Confluence docs.
Changelog doesn't need to explain every single day-to-day change within the branch; the commit history should make that clear. The changelog should detail what changes the branch as a whole is introducing to the branch that it is being pulled in to.
I have not looked that closely at the code, if there are blocks that you want me to sanity check for you please point me to them and I'll take a closer look :)
The only change being requested is to modularize and rename the data-parsing functions in the EDA notebook because they add some clutter to what could otherwise be a pretty short notebook.
However please also wait for @amanjeev 's review before merging this PR, for I have not built this PR.

**Review by yuriy-pavlov — APPROVED** [review]

**Pull request merged**
